modules/Trackers/Tracker.php

```diff
@@ -110,14 +110,11 @@ function get_recently_viewed($user_id, $modules = '')
 			$breadCrumb = $_SESSION['breadCrumbs'];
 	        $module_query = '';
 	        if(!empty($modules)) {
-	           $history_max_viewed = 10;
 	           $module_query = is_array($modules) ? ' AND module_name IN (\'' . implode("','" , $modules) . '\')' :  ' AND module_name = \'' . $modules . '\'';
-	        } else {
-	           $history_max_viewed = (!empty($GLOBALS['sugar_config']['history_max_viewed']))? $GLOBALS['sugar_config']['history_max_viewed'] : 50;
 	        }
 
-	        $query = 'SELECT item_id, item_summary, module_name, id FROM ' . $this->table_name . ' WHERE id = (SELECT MAX(id) as id FROM ' . $this->table_name . ' WHERE user_id = \'' . $user_id . '\' AND deleted = 0 AND visible = 1' . $module_query . ')';
-	        $result = $this->db->limitQuery($query,0,$history_max_viewed,true,$query);
+	        $query = 'SELECT item_id, item_summary, module_name, id FROM ' . $this->table_name . ' WHERE user_id = \'' . $user_id . '\' AND deleted = 0 AND visible = 1' . $module_query . ' ORDER BY id DESC LIMIT 1';
```

Comment:
shouldn't remove limitQuery and shouldn't add LIMIT due to MSSQL. But I think perhaps the select MAx can be removed...

Reply:
Ah, ok. Did not think about mssql. I just thought query should be faster than limitQuery so it was an additional optimization. But you are right maybe using a limitQuery with limit 1 is the better way to go.

Reply:
You can. maybe, use something like:

switch ($db_type) {
  case 'mysql':
    $query = .... // Your  proposed query
    break;
  case 'mssql':
    $query = "SELECT .... ORDER BY id DESC OFFSET 0 ROWS FETCH FIRST ROW ONLY" ;
    break;
}

This gives the correct syntax for the query, allowing for future expansion...

Reply:
Can you check again if the new patch is ok with mssql?